homeassistant/components/nobo_hub/select.py

```diff
@@ -0,0 +1,169 @@
+    OVERRIDE_TYPE_NOW,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up any temperature sensors connected to the Nobø Ecohub."""
+
+    # Setup connection with hub
+    hub: nobo = hass.data[DOMAIN][config_entry.entry_id]
+
+    override_type = (
+        nobo.API.OVERRIDE_TYPE_NOW
+        if config_entry.options.get(CONF_OVERRIDE_TYPE) == OVERRIDE_TYPE_NOW
+        else nobo.API.OVERRIDE_TYPE_CONSTANT
+    )
+
+    entities: list[SelectEntity] = [
+        NoboProfileSelector(zone_id, hub) for zone_id in hub.zones
+    ]
+    entities.append(NoboGlobalSelector(hub, override_type))
+    async_add_entities(entities, True)
+
+
+class NoboGlobalSelector(SelectEntity):
+    """Global override selector for Nobø Ecohub."""
+
+    _attr_has_entity_name = True
+    _attr_translation_key = "global_override"
+    _attr_device_class = "nobo_hub__override"
+    _attr_should_poll = False
+    _modes = {
+        nobo.API.OVERRIDE_MODE_NORMAL: "none",
+        nobo.API.OVERRIDE_MODE_AWAY: "away",
+        nobo.API.OVERRIDE_MODE_COMFORT: "comfort",
+        nobo.API.OVERRIDE_MODE_ECO: "eco",
+    }
+    _attr_options = list(_modes.values())
+    _attr_current_option: str
+
+    def __init__(self, hub: nobo, override_type) -> None:
+        """Initialize the global override selector."""
+        self._nobo = hub
+        self._attr_unique_id = f"{hub.hub_serial}"
+        self._override_type = override_type
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, f"{hub.hub_serial}")},
+            name=hub.hub_info[ATTR_NAME],
+            manufacturer=NOBO_MANUFACTURER,
+            model=f"Nobø Ecohub ({hub.hub_info[ATTR_HARDWARE_VERSION]})",
+            sw_version=hub.hub_info[ATTR_SOFTWARE_VERSION],
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Register callback from hub."""
+        self._nobo.register_callback(self._after_update)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Deregister callback from hub."""
+        self._nobo.deregister_callback(self._after_update)
+
+    async def async_select_option(self, option: str) -> None:
+        """Set override."""
+        mode = [k for k, v in self._modes.items() if v == option][0]
+        try:
+            await self._nobo.async_create_override(
+                mode, self._override_type, nobo.API.OVERRIDE_TARGET_GLOBAL
+            )
+        except Exception as exp:
+            raise HomeAssistantError from exp
+
+    async def async_update(self) -> None:
+        """Fetch new state data for this zone."""
+        self._read_state()
+
+    @callback
+    def _read_state(self) -> None:
+        for override in self._nobo.overrides.values():
+            if override["target_type"] == nobo.API.OVERRIDE_TARGET_GLOBAL:
+                self._attr_current_option = self._modes[override["mode"]]
```

Comment:
shouldn't we break out of the for loop?
```suggestion
                self._attr_current_option = self._modes[override["mode"]]
                break
```

Reply:
Done